### PR TITLE
feat: per-adapter configuration for tool use and thinking budget

### PR DESCRIPTION
**Pull request by pacaplan**
## Summary

Implements the [add-adapter-configuration](openspec/changes/add-adapter-configuration/) OpenSpec change proposal, enabling per-adapter `allow_tool_use` and `thinking_budget` settings in `.gauntlet/config.yml`.

### Changes

- **Schema**: Added `adapterConfigSchema` with `allow_tool_use` (boolean, default true) and `thinking_budget` (enum off/low/medium/high, optional). Extended `cliConfigSchema` with optional `adapters` record.
- **Thinking budget maps**: Created `src/cli-adapters/thinking-budget.ts` with level-to-value mappings for Claude (MAX_THINKING_TOKENS), Codex (model_reasoning_effort), and Gemini (settings.json thinkingBudget).
- **Claude adapter**: Dynamic `--tools ""`/`--allowedTools` args based on `allowToolUse`, `MAX_THINKING_TOKENS` env var for thinking budget.
- **Codex adapter**: `--disable shell_tool` for tool use control, `-c model_reasoning_effort` for thinking budget.
- **Gemini adapter**: Conditional `--allowed-tools` for tool use, temporary `.gemini/settings.json` manipulation with serialized lock to prevent race conditions in parallel execution.
- **Config threading**: Adapter configs flow from `runner.ts` → `review.ts` → `adapter.execute()`, looked up by adapter name.
- **Tests**: 26 new tests covering schema validation, thinking budget maps, config threading, and Gemini settings.json backup/restore.

### Config example

```yaml
cli:
  default_preference: [claude, codex, gemini]
  adapters:
    claude:
      thinking_budget: high
    codex:
      thinking_budget: high
    gemini:
      thinking_budget: high
```

## Test plan

- [x] All 543 tests pass (26 new)
- [x] TypeScript build passes
- [x] Biome lint passes
- [x] Security scan passes
- [x] Schema validation passes
- [x] Code quality reviews pass
- [x] Change review passes
- [x] Task review passes

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Project-level per-adapter settings in .gauntlet/config.yml to control tool usage and thinking budgets (off/low/medium/high) for Claude, Codex, and Gemini; runtime now honors per-adapter allow_tool_use and thinking_budget (with sensible defaults).

* **Documentation**
  * Added design, proposal, and spec docs describing schema, mappings, and behavior/edge-cases for adapter controls.

* **Tests**
  * Added unit and integration tests for schema validation, budget mappings, option threading, and Gemini settings handling.

### CODE REVIEW

File: .gauntlet/config.yml [hunk 1]
```diff
@@ -10,6 +10,13 @@ cli:
     - claude
     - codex
     - gemini
+  adapters:
```
Comment (pacaplan) [inline .gauntlet/config.yml]: add explicit allow_tool_use setting to each one, with true for claude and codex, false for gemini
